### ISSUE TITLE
mode collapse?

### ISSUE THREAD

**Issue by dribnet** [carpedm20/BEGAN-tensorflow#10]
Hi @carpedm20 - this codebase is now working well for me, wanted to compare notes. with your latest changes I get pretty good results on 64x64 celeba. 128x128 doesn't do as well - probably needs different hyperparams.

Here are some 64x64 results after 130k cycles of training. first x_fixed, D_real :

![x_fixed](https://cloud.githubusercontent.com/assets/945979/25028026/4a94cf6a-2105-11e7-89ad-297f5ab56f90.png)

![130000_d_real](https://cloud.githubusercontent.com/assets/945979/25028036/66f51070-2105-11e7-8ef8-01779af7ab9a.png)

That looks like pretty good auto-encoder reconstruction results - the resemblance is pretty clear. OK, now here's the D_fake, and G:

![130000_d_fake](https://cloud.githubusercontent.com/assets/945979/25028072/d06a100a-2105-11e7-9d2d-f6b1b5e23a13.png)

![130000_g](https://cloud.githubusercontent.com/assets/945979/25028076/d713f826-2105-11e7-862a-6ce5aa6449f7.png)

Now here my knowledge of this model gets a little hazy. These look fairly good and they look like each other - but they no longer bear very much resemblance x_fixed - should they? If so, then I think this is would likely be a symptom of a pretty significant mode collapse.

Just for comparison, here's what taking the same x_fixed looks like when passed through a trained [ALI model](https://ishmaelbelghazi.github.io/ALI/):

![x_fixed](https://cloud.githubusercontent.com/assets/945979/25028237/ff17d0c6-2106-11e7-8a63-013781545261.png)

![20170414_ali64_montage_began](https://cloud.githubusercontent.com/assets/945979/25028262/2542ebe6-2107-11e7-8bdc-9de7bd60e9a6.png)

Some of these come out better than others, but note the overall variation in hair color, skin tone, glasses, hats, backgrounds that seems to be missing from the current BEGAN generator.  Curious if you have any thoughts from your own investigations.

**Comment by carpedm20** [comment]
Maybe authors can give more explanation on this and I don't know much about ALI, I guess the D that BEGAN uses is l1 loss between auto-encoder input and output. G tries to mimicking smoothed output of auto-encoder so it leads to smoothed, same-looking faces.

Authors mentioned that simply reducing the learning rate was helpful to get out from mode collapse so I guess changing the learning rate schedule after 100k might be helpful.

**Comment by dribnet** [comment]
Thanks for the feedback. I'll close and might try out playing with the learning rate.
